### ISSUE TITLE
Improve test procedures

### ISSUE THREAD

**Issue by kkohbrok** [openmls/openmls#1543]
The way we run tests is currently somewhat fragmented. Some tests only run on the RustCrypto provider and ciphersuites are hardcoded in some places. Also, the current `rstest`-based tests have hardcoded ciphersuites (as opposed to running all ciphersuites provided by a given provider).

Tests could be better in a few ways:

- Tests only need to iterate over crypto providers, not the associated storage providers 
- All test should iterate over all providers (not just RustCrypto)
- All test should iterate over all ciphersuites supported by the provider (instead of hard-coding them)
- It should be possible to create several storage provider instances inside of a test (e.g. to instantiate multiple clients)

This is related to the issue that OpenMLS internally maintains an enum of Ciphersuites instead of treating them as an integer and letting the crypto provider determine if they support a given ciphersuite.

**Comment by franziskuskiefer** [comment]
I made a proc-macro for some of that a while ago, but never merged it because it was part of the libcrux work. https://github.com/openmls/openmls/blob/a4fb8358a87effd8a48ac0c114e6de386e699aa4/openmls_test/Readme.md
We could re-use that code to build something like this.

**Comment by kkohbrok** [comment]
Yes, that looks like it might do the trick. It remains the issue on how to instantiate multiple clients with distinct storage providers in a given test. I don't think that's strictly necessary right now, but it might be in the future, e.g. if we store group state indexed by a group id.

**Comment by keks** [comment]
On first look this seems to be resolved by the PR, only leaving the followup issue #1574. Is that right and can we close this one @franziskuskiefer?

**Comment by franziskuskiefer** [comment]
> It should be possible to create several storage provider instances inside of a test (e.g. to instantiate multiple clients)

This is still open. But please update the description to reflect that most things are done and what's still open.
